pkg/connector/inet/inet.go

```diff
@@ -155,26 +154,24 @@ func (c *Connection) SendFleetAPICommand(ctx context.Context, endpoint string, c
 
 // Connection implements the connector.Connector interface by POSTing commands to a server.
 type Connection struct {
-	UserAgent  string
-	vin        string
-	client     http.Client
-	serverURL  string
-	inbox      chan []byte
-	authHeader string
+	UserAgent string
+	vin       string
+	client    *http.Client
+	serverURL string
+	inbox     chan []byte
 
 	wakeLock sync.Mutex
 	lastPoke time.Time
 }
 
 // NewConnection creates a Connection.
-func NewConnection(vin string, authHeader, serverURL, userAgent string) *Connection {
+func NewConnection(vin string, client *http.Client, serverURL, userAgent string) *Connection {
```

Comment:
The requirements for `client` need to be clearly documented, ideally with an example of setting up a token source. Alternatively, we could create a `NewConnectionWithClient(client *http.Client, vin string ...) *Connection` function (with documentation + example).